nsqd/protocol_v2_test.go

```diff
@@ -1452,6 +1453,83 @@ func runAuthTest(t *testing.T, authResponse, authSecret, authError, authSuccess
 
 }
 
+type fakeNetConn struct {
+	read             func([]byte) (int, error)
+	write            func([]byte) (int, error)
+	closeFunc        func() error
+	localAddr        func() net.Addr
+	remoteAddr       func() net.Addr
+	setDeadline      func(time.Time) error
+	setReadDeadline  func(time.Time) error
+	setWriteDeadline func(time.Time) error
+}
+
+func (f fakeNetConn) Read(b []byte) (n int, err error)   { return f.read(b) }
+func (f fakeNetConn) Write(b []byte) (n int, err error)  { return f.write(b) }
+func (f fakeNetConn) Close() error                       { return f.closeFunc() }
+func (f fakeNetConn) LocalAddr() net.Addr                { return f.localAddr() }
+func (f fakeNetConn) RemoteAddr() net.Addr               { return f.remoteAddr() }
+func (f fakeNetConn) SetDeadline(t time.Time) error      { return f.setDeadline(t) }
+func (f fakeNetConn) SetReadDeadline(t time.Time) error  { return f.setReadDeadline(t) }
+func (f fakeNetConn) SetWriteDeadline(t time.Time) error { return f.setWriteDeadline(t) }
+
+type fakeNetAddr struct{}
+
+func (fakeNetAddr) Network() string { return "" }
+func (fakeNetAddr) String() string  { return "" }
+
+func newFakeNetConn() fakeNetConn {
+	netAddr := fakeNetAddr{}
+	return fakeNetConn{
+		read: func(b []byte) (int, error) {
+			return copy(b, []byte("INVALID_COMMAND\n")), nil
+		},
+		localAddr: func() net.Addr {
+			return netAddr
+		},
+		remoteAddr: func() net.Addr {
+			return netAddr
+		},
+		setDeadline:      func(time.Time) error { return nil },
+		setWriteDeadline: func(time.Time) error { return nil },
+		setReadDeadline:  func(time.Time) error { return nil },
+		closeFunc:        func() error { return nil },
+	}
+}
```

Comment:
I'll try it out and remove the `read` implementation, leaving that for the individual `Test` functions.